### PR TITLE
MAINT: Temporarily change the default RNG in check_random_state

### PR DESCRIPTION
**Pull request by bashtage**
Change to the singleton generator for the 0.13 release only
Warn the user

- [X] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by bashtage** [comment]
@tupui I have temporarily defaulted to the singleton generator in this. This should let sm work with older scipy, but it warns that the behavior will change after 0.13 and tells users what to do instead.

**Comment by bashtage** [comment]
@tupui More importantly, I couldn't get QMC to work for me here.  Does it work? I think QMC doesn't do real multivariate drawns, e.g., 

Halton(1).random((10,2)) doesn't work.  Similarly Halton(2).random((10,2)) also doesn't work. 

Can you get QMC to work?

**Comment by tupui** [comment]
> @tupui More importantly, I couldn't get QMC to work for me here. Does it work? I think QMC doesn't do real multivariate drawns, e.g.,
> 
> Halton(1).random((10,2)) doesn't work. Similarly Halton(2).random((10,2)) also doesn't work.
> 
> Can you get QMC to work?

The `random` does not take a shape but a number of samples. So if you want 2 dimensions you would do `Halton(d=2).random(n=10)`. The methods in the QMC module are all working like this and are multivariate.

We did not want to have `shape` as we are already passing the dimension when instantiating the class. It would just be redundant to have it in the sampling part and also would be user error prone.

**Comment by bashtage** [comment]
That is what I thought but I thinknit doesn't work with copulas for the same reason. Might need special treatment.

**Comment by tupui** [comment]
> That is what I thought but I thinknit doesn't work with copulas for the same reason. Might need special treatment.

Oh, I thought the tests were passing. What is not working?

File: statsmodels/tools/rng_qrng.py [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 import scipy.stats as stats
 
+_future_warn = """\
+Passing `None` as the seed currently return the NumPy singleton RandomState
+(np.random.mtrand._rand). After release 0.13 this will change to using the
+default generator provided by NumPy (np.random.default_rng()). If you need
+reproducible draws, you should pass a seeded np.random.Generator, e.g.,
+
+import numpy as np
+seed = 32839283923801
```
Comment (tupui) [inline statsmodels/tools/rng_qrng.py:11]: Maybe it could help to explain that a good quality seed can be generated with `np.random.SeedSequence().entropy`

**Review by tupui — APPROVED** [review]

**Comment by josef-pkt** [comment]
If qmc doesn't use the same pattern for shape as scipy distributions rvs and numpy, then it will be messy to write code that works will all 3.

The archimedean copulas have currently rvs through the mixture representation, which has not good properties with qmc.

**Comment by tupui** [comment]
> If qmc doesn't use the same pattern for shape as scipy distributions rvs and numpy, then it will be messy to write code that works will all 3.
> 
> The archimedean copulas have currently rvs through the mixture representation, which has not good properties with qmc.

IMO it's just a wrapper to have on `check_random_state` if it's getting a `QMCEngine`. Or we add a dedicated function to sample using QMC in the copula base class. We have this in SciPy's [NumericalInverseHermite](https://scipy.github.io/devdocs/reference/generated/scipy.stats.NumericalInverseHermite.html).

**Comment by josef-pkt** [comment]
The main addition that we have to add to copulas outside elliptical is to have inverse sampling method for rvs generation, i.e. conditional distributions/ppf. Then we would have to switch to that if random state is a qmc.

**Comment by tupui** [comment]
@bashtage @josef-pkt let me know what you want me to do. I see two options right now:
1. wrap `QMCEngine` to just use the first element of the shape as `n` and ensure `QMCEngine.d` matches the second element.
2. add another method for QMC sampling. We have `random` we could either have a `qmc_engine` parameter or another `qmc_random` (or any other name).

I can do a PR or this could be in this one. As you wish.

**Pull request merged**